src/pdf_extract.py

```diff
@@ -0,0 +1,45 @@
+#
+# This module contains support functions which extract text and image objects
+# from a PDF file for inclusion in other media.
+#
+from typing import Any, Union
+
+from PIL.PngImagePlugin import ImageFile
+from pypdf import PdfReader
+
+PDF_ITEMS = dict[str, Union[list[str], dict[str, ImageFile]]]
+
+
+def get_pdf_objects(
+    config: dict[str, dict[str, Any]]
+) -> dict[str, dict[str, PDF_ITEMS]]:
+    """Extract all images and text from the configured PDF files.
+
+    Assumes that the configuration reflects the following YAML:
+        teams:
+          Practices Team:
+            team_dashboards:
+              Developer Practices Dashboard:
+                Filename: DPROD_Report.pdf
+              ...
+          ...
+        config:
+          Download_path: "downloaded_pdfs/"
+    """
+    results = {}
+    for team, dashboards in config["teams"].items():
+        team_results = {}
+        for dashboard, options in dashboards["team_dashboards"].items():
+            dashboard_results = {"images": {}, "text": []}
+            pdf = PdfReader(config["config"]["Download_path"] + options["Filename"])
+            for page in pdf.pages:
+                i = {image.name: image.image for image in page.images}
```

Comment:
I generated a new test PDF which contains multiple pages.  It turns out that the images are labeled sequentially through the whole document.  (I _think_ (but did not confirm) that this is a PDF requirement, because a document is actually a structured collection of objects, each of which is individually addressable, so having a repeated name would cause problems internally as well as for our purposes.)

So, I'm not inclined to be paranoid here.  In the future, if we hit a case where an image is missing from the output, we can revisit this code.